src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java

```diff
@@ -15,7 +16,9 @@
 
     private static final double INCREASE_TIMESTAMP = 0.0000000001D;
     private static RoadCapacityAdjustmentFunction roadCapacityAdjustmentFunction;
-    private final HashMap<Vehicle,Double> caccShareEncounteredByVehicle=new HashMap<>();
+    private static AdditionalLinkTravelTimeCalculationFunction additionalLinkTravelTimeCalculationFunction;
```

Comment:
This should be an `Optional` with the default being `Optional.empty`  and then use a `map` to calculate followed by `orElse` returning `0`